### DISCUSSION TITLE
Enable group runners

### DISCUSSION THREAD

**Discussion by KameKam** [python-gitlab/python-gitlab#1502] [Q&A] [answered]
Hello
I'm making a python script to create a project with a import of a template  project, until now everything is ok. When I try to run a pipeline through script, it's stuck because group runners are disabled by default in our gitlab. I seek many site and docs to find solution but no result or maybe i used them wrong.
![gitlab-group_runners](https://user-images.githubusercontent.com/49527647/120893574-48ebf180-c614-11eb-91ab-ef8cd5029473.png)
I found this solution but got 403 response, while i want just enable group runners like on image : 
```
import gitlab

# gl is the Gitlab object
gl = gitlab.Gitlab.from_config('pvcp', ['./.python-gitlab.cfg'])
project=gl.projects.get(301)

print(project.runners.list())
p_runner = project.runners.create({'runner_id': 20})`

PS C:XXXXX\Desktop\Python-Gitlab> .\main2.py
[<ProjectRunner id:4>, <ProjectRunner id:20>]
Traceback (most recent call last):
  File "C:\Python39\lib\site-packages\gitlab\exceptions.py", line 287, in wrapped_f
    return f(*args, **kwargs)
  File "C:\Python39\lib\site-packages\gitlab\mixins.py", line 321, in create
    server_data = self.gitlab.http_post(path, post_data=data, files=files, **kwargs)
  File "C:\Python39\lib\site-packages\gitlab\client.py", line 713, in http_post
    result = self.http_request(
  File "C:\Python39\lib\site-packages\gitlab\client.py", line 584, in http_request
    raise gitlab.exceptions.GitlabHttpError(
gitlab.exceptions.GitlabHttpError: 403: 403 Forbidden - Runner is a group runner

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\XXXXX\Desktop\Python-Gitlab\main2.py", line 8, in <module>
    p_runner = project.runners.create({'runner_id': 20})
  File "C:\Python39\lib\site-packages\gitlab\exceptions.py", line 289, in wrapped_f
    raise error(e.error_message, e.response_code, e.response_body) from e
gitlab.exceptions.GitlabCreateError: 403: 403 Forbidden - Runner is a group runner
```

Thanks a lot

**Comment by JohnVillalovos** [comment, marked as answer]
TL;DR: I believe this is a Gitlab API issue and not something that python-gitlab can fix.

It seems like group runners can only be enabled or disabled as a group. Meaning either the project can use the group's runners or not.

It does not appear that a single group runner can be enabled. Only all group runners are enabled or all group runners are disabled.

An individual project runner can be enabled or disabled, but that is not what you have there.

I don't see any support in the API to enable/disable group runners per project 😟 But maybe I don't see it.

This is what generates that error you see:
https://gitlab.com/gitlab-org/gitlab/-/blob/c9a19a219fb7fd84fc2c321a0de5adb54049365d/lib/api/ci/runners.rb#L274

**Comment by KameKam** [comment]
Thanks a lot for your answer, you even find the code bloc with my errors :)  It's confirm my first think., there is for moment no support for this feature in api.
